packages/netlify-cms-core/src/actions/config.js

```diff
@@ -61,9 +62,37 @@ export function applyDefaults(config) {
           }
         }),
       );
+
+      // Set default value function if widget provides a static createDefaultValue method
+      map.set(
+        'collections',
+        map.get('collections').map(collection => {
+          const fields = collection.get('fields');
+          if (fields) {
+            return collection.set('fields', bindCreateDefaultValues(fields))
+          }
+          return collection;
+        })
+      );
     });
 }
 
+function bindCreateDefaultValues(fields) {
+  return fields.map(field => {
+    return field.withMutations(field => {
+      if (!field.get('default')) {
+        const createDefaultValue = resolveWidget(field.get('widget')).control.createDefaultValue;
+        if (createDefaultValue) {
+          field.set('default', createDefaultValue);
+        }
+      }
+      if (field.get('fields')) {
```

Comment:
@ianhomer I think you also need to account for `field` widgets

Reply:
Yeah I was thinking the same - probably types too, right?

Reply:
Yes, `types` as well.